tests/chainerx_tests/conftest.py

```diff
@@ -21,11 +21,13 @@ def pytest_runtest_teardown(item, nextitem):
 def pytest_generate_tests(metafunc):
     marker = [
         m for m in metafunc.definition.iter_markers()
-        if m.name == 'parametrize_device']
+        if m.name in ('parametrize_device', 'parametrize_device_name')]
     if marker:
         marker, = marker  # asserts len == 1
         device_names, = marker.args
-        metafunc.parametrize('device', device_names, indirect=True)
+        # fixture name is either device or device_name
+        fixture_name = marker.name.split('_', 1)[1]
```

Comment:
This code looks too adhoc and vulnerable.
It's better to make it explicit, like writing `if marker.name == ... : fixture_name = ...` blocks, or using a dict.